### PR TITLE
fix: maps getting lat/lon from s2_cell_id or fields/tags

### LINKED ISSUE #997: Test Maps: More link is broken
After giraffe issue is fixed, pull in the latest giraffe version and test locally. Link to giraffe issue: https://github.com/influxdata/giraffe/issues/519

This link needs to be updated. Let's use this link when it is live: https://docs.influxdata.com/influxdb/cloud/visualize-data/visualization-types/map/

image

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #997 Closes #giraffe/523

Updated the function to get Lat Lon such that it will not render giraffe plot if the correct data is not provided. Added a condition to get `s2_cell_id` or `lat`/`lon` as fields/tags and throw an error if none of those exist in the data provided

### CODE REVIEW

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 49 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1150)

File: src/shared/utils/vis.ts [hunk 1]
```diff
@@ -332,12 +399,44 @@ const getColumnValue = (table: Table, field: string) => {
...
+      }
+    case 'lat_lon_as_fields':
+      const latCoordinate = getColumnValue(table, 'lat')
+      const lonCoordinate = getColumnValue(table, 'lon')
+      return {
+        lat: parseCoordinates(latCoordinate),
+        lon: parseCoordinates(lonCoordinate),
+      }
+    default:
+      throw new Error('lat_lon_not_provided')
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:426]: :+1: i think throwing an error here is the best way to handle this, nice

File: src/visualization/types/Map/view.tsx [hunk 2]
```diff
@@ -61,52 +71,53 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
...
+  } else if (coordinateError === MapErrorStates.InvalidCoordinates) {
+    error =
+      'Map type is not supported with the data provided: Missing latitude/longitude values (field values must be specified to lat or lon)'
+
+    return (
+      <div className="panel-resizer--error" data-testid="geoplot-error">
+        {error}
+      </div>
+    )
+  } else {
```
Comment (hoorayimhelping) [inline src/visualization/types/Map/view.tsx:94]: ✂️ the `else` - this is the main 'happy path' of the code. it should be the first level of indentation, not squirreled away in an `else` statement. those `return` statements in the `if` statements make `else` (and `else if`) unnecessary

**Review by hoorayimhelping — CHANGES_REQUESTED** [review]
Good work! A couple of things, nothing major. Mostly about using enums or constants instead of strings

File: src/shared/utils/vis.ts [hunk 3]
```diff
@@ -314,6 +315,72 @@ export const getMainColumnName = (
...
+const getCoordinateColumn = (table: Table): string => {
+  const column = table.getColumn('s2_cell_id')
+  if (column != null) {
+    return 's2_cell_id'
+  }
+  const lat = table.getColumn('lat')
+  const lon = table.getColumn('lon')
+
+  if (lat != null && lon != null) {
+    return 'lat_lon_as_tags'
```
[the remark this answers is not in the document]
Comment (Palakp41) [reply src/shared/utils/vis.ts:327]: Good idea!

File: src/shared/utils/vis.ts [hunk 4]
```diff
@@ -314,6 +315,83 @@ export const getMainColumnName = (
...
+
+    const latCoordinate = getColumnValue(table, 'lat')
+    const lonCoordinate = getColumnValue(table, 'lon')
+
+    if (latCoordinate && lonCoordinate) {
+      return CoordinateType.Fields
+    }
+
+    return CoordinateType.None
+  } catch (e) {
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:346]: if lint doesn't complain, it's probably fine. but i think you can just do:

```ts
catch {
  throw new Error('
}

File: src/visualization/types/Map/view.tsx [hunk 5]
```diff
@@ -61,53 +76,60 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
...
-    default:
-      const tileServerConfiguration = {
-        tileServerUrl: getMapboxUrl(),
-        bingKey: '',
-      }
+  if (
+    mapServiceError === RemoteDataState.NotStarted ||
+    coordinateError === RemoteDataState.NotStarted ||
+    mapServiceError === RemoteDataState.Loading ||
+    coordinateError === RemoteDataState.Loading
```
Comment (hoorayimhelping) [inline src/visualization/types/Map/view.tsx:83]: :+1:

File: src/visualization/types/Map/view.tsx [hunk 6]
```diff
@@ -61,53 +76,60 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
...
+
+    return (
+      <div className="panel-resizer--error" data-testid="geoplot-error">
+        {error}
+      </div>
+    )
+  }
+  const tileServerConfiguration = {
+    tileServerUrl: getMapboxUrl(),
+    bingKey: '',
```
Comment (hoorayimhelping) [inline src/visualization/types/Map/view.tsx:109]: is this a holdover from bonitoo's work?

**Review by hoorayimhelping — APPROVED** [review]

File: src/visualization/types/Map/view.tsx (hunk 6 above)
Comment (Palakp41) [reply src/visualization/types/Map/view.tsx:109]: So if bingKey is provided, it renders bing maps. If not, it renders the map url we send it.

Comment (hoorayimhelping) [reply src/visualization/types/Map/view.tsx:109]: fancy!

**Pull request merged**